### PR TITLE
added site stream time checks

### PR DESCRIPTION
**Pull request by ozym**
Adds some extra logging for station / site / stream time span overlaps.

Currently it's just a log message (go test -v) but once the backlog is cleared it should be set to a full Error.

/cc @nbalfour @salichon

### CODE REVIEW

File: tests/sites_test.go [hunk 1]
```diff
@@ -21,4 +32,29 @@ func TestSites(t *testing.T) {
 		}
 	})
 
+	t.Run("check for invalid site dates", func(t *testing.T) {
```
Comment (quiffman) [inline tests/sites_test.go]: Suggestion:
> "check for invalid dates; site within station"

File: tests/streams_test.go [hunk 2]
```diff
@@ -102,6 +102,31 @@ func TestStreams(t *testing.T) {
 		}
 	})
 
+	t.Run("check for invalid stream station dates", func(t *testing.T) {
```
Comment (quiffman) [inline tests/streams_test.go]: Suggestion:
> "check for invalid dates; stream within station"

File: tests/streams_test.go [hunk 3]
```diff
@@ -120,6 +145,33 @@ func TestStreams(t *testing.T) {
 		}
 	})
 
+	t.Run("check for invalid stream site dates", func(t *testing.T) {
```
Comment (quiffman) [inline tests/streams_test.go]: Suggestion:
> "check for invalid dates; stream within site"

**Comment by nbalfour** [comment]
@quiffman that makes sense to me

**Comment by ozym** [comment]
@quiffman have updated check labels and added a REAME.md file in the tests directory. There is still some tidy up work needed in the test files (style changed over time as new go features were released).

I've needed to add a "delta.go" file to the directory to keep the latest version of the go testing framework happy.

**Comment by salichon** [comment]
@quiffman 
A bit late but that makes sens to have these tests :)
Having these dates flagged helps in checking that they re matching the data too.

**Pull request merged**